src/batched-json.js

```diff
@@ -48,12 +49,26 @@ export function batchFetchJsonFor(
   element,
   opt_expr = '.',
   opt_urlReplacement = UrlReplacementPolicy.NONE,
-  opt_refresh = false)
+  opt_refresh = false,
+  opt_token = undefined)
 {
   assertHttpsUrl(element.getAttribute('src'), element);
   const xhr = Services.batchedXhrFor(ampdoc.win);
   return requestForBatchFetch(element, opt_urlReplacement, opt_refresh)
-      .then(data => xhr.fetchJson(data.xhrUrl, data.fetchOpt))
+      .then(data => {
+        const crossOriginAttr = element.getAttribute('cross-origin');
+        if (crossOriginAttr &&
+            crossOriginAttr.indexOf('amp-viewer-auth-token-via-post') >= 0) {
```

Comment:
Can we just check `if (opt_token)` instead of inspecting the attribute value again?

Reply:
Yes, given the token should only be coming from this flow (where the attribute is present), that should be fine.